pages/map.tsx

```diff
@@ -38,46 +40,54 @@ function Map() {
     setMap(map);
     setUsername(user?.name!);
   },[]);
- 
+  
+  const { interestZone, setInterestZone } = useContext(InterestZoneProviderContext)
+
   useEffect(()=>{
     zoneServiceUi.findAll().then((zones)=>{
       setInterestZones(zones);
     })   
   },[map])
 
+  useEffect(()=>{
+    if (interestZone)
+      setEditModalVisible(true);
+  },[interestZone])
```

Comment:
Please remove the above hook & associated state, and add just before the template the following logic
`const editModalVisible = (zone !== undefined);`